### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by porag-m06**
#Highlights
- `Person` class created.
- `Student` class created.
- `Teacher` class created.
- Each `class` saved in sep

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @porag-m06 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- Kindly include the Future Features section in your README file. The section should include 1-3 features that can be implemented in the future. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+
+# The Student class represents a student with an id, name, age, and classroom.
+# It inherits from the Person class and includes an additional method to play hooky.
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (thecodechaser) [inline student.rb]: - [ ] Please make sure to pass the parameters in the super method in the correct order as the parent class initializer expects.
  - You can consider the below code.
  
```ruby
super(age, name, parent_permission)

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+
+# The Teacher class represents a teacher with a name, age, and specialization.
+# It inherits from the Person class and overrides the method can_use_services?.
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (thecodechaser) [inline teacher.rb]: - [ ] Please make sure to pass the parameters in the super method in the correct order as the parent class initializer expects.
  - You can consider the below code.
  
```ruby
super(age, name, parent_permission)

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @porag-m06👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with classes. ✔️ 
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- Person class created. ✔️ 
- Student class created. ✔️
- Teacher class created. ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @iambenkis in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**